plugins/uefi-capsule/efi/generate_binary.py

```diff
@@ -95,6 +93,17 @@ def _run_objcopy(args):
             ".dynsym",
             "-j",
             ".rel*",
+        ]
+        if args.sbat_distro_id:
+            argv += [
+                "-j",
+                ".sbat",
+                "--set-section-alignment",
+                ".sbat=512",
+                "--change-section-vma",
+                ".sbat=0x2000",
```

Comment:
same for this to not be a magic value, i.e:

```
# The gnu-efi linker script efi.lds places the .text section at the
# virtual memory address 0x4000. Place the .sbat section below this.
```

Reply:
I don't think the linker script specifies the virtual address of the .text section though - it only specifies page alignment. I think the address is determined by the linker based on that alignment and the size of the preceding sections. In the example in https://github.com/fwupd/fwupd/pull/2914#issuecomment-782780063, the .text section would be placed at 0x3000 if the .eh_frame section were 284 bytes smaller.

Reply:
@chrisccoulson is there a fixed address that you think could be safe to use? Since as you mentioned the sections are page aligned by the `efi.lds` linker script and there are a couple of sections before `.text` (that are discarded when creating the PE binary), so at the very least `.text` would be above `0x1000` IIUC.